test/OpenTelemetry.Tests/Logs/LogRecordTest.cs

```diff
@@ -606,6 +606,39 @@ public void ParseStateValuesUsingCustomTest()
             }
         }
 
+        [Fact]
+        public void DisposingStateTest()
+        {
+            this.options.ParseStateValues = true;
+            try
+            {
+                DisposingState state = new DisposingState("Hello world");
+
+                this.logger.Log(
+                    LogLevel.Information,
+                    0,
+                    state,
+                    null,
+                    (s, e) => "OpenTelemetry!");
+                var logRecord = this.exportedItems[0];
+
+                state.Dispose();
+
+                Assert.Null(logRecord.State);
+                Assert.NotNull(logRecord.StateValues);
+                Assert.Equal(1, logRecord.StateValues.Count);
+
+                KeyValuePair<string, object> actualState = logRecord.StateValues[0];
+
+                Assert.Same("Value", actualState.Key);
+                Assert.Same("Hello world", actualState.Value);
+            }
+            finally
+            {
+                this.options.ParseStateValues = false;
```

Comment:
I think all these `finally` to reset the `this.option` are not needed as `this.option` should be set to a new value for each test (as it is assigned in the constructor)

Reply:
Updated